Advanced Transformations beta updates (#3874)
segmentio/segment-docs: src/protocols/transform/index.md

## What is a Transformation?

> With Transformations, you can change data as it flows through Segment to either correct bad data or customize data for a specific destination. Change event and property names to align events with your Tracking Plan, or to conform to a destination-specific requirement. For example, a Transformation could be created to change an event name from `completed_order` to `Order Completed` to conform to [Segment's ecommerce spec](/docs/connections/spec/ecommerce/v2/#order-completed).

Reviewer: ```suggestion
With Transformations, you can change data as it flows through Segment to either correct bad data or customize data for a specific destination. Change event and property names to align events with your Tracking Plan, or to conform to a destination-specific requirement. For example, you can create a Transformation to change an event name from `completed_order` to `Order Completed` to conform to [Segment's ecommerce spec](/docs/connections/spec/ecommerce/v2/#order-completed).
```

Let's make this active voice.

### Other important notes

- **Transformations cannot be applied retroactively:** They only apply to data moving forward. However, you can manually extract and re-send (or even [Replay](/docs/guides/what-is-replay)) events through a source with an active destination Transformation, which will send the transformed events to your destinations.

> - **Transformations are only available to Protocols customers:** If you're interested in this feature, contact your Account Executive or CSM to learn more about the Protocols package.

Reviewer: ```suggestion
- **Transformations are available to Protocols customers:** If you're interested in this feature, contact your Account Executive or CSM to learn more about the Protocols package.
```

> Workspace Owner or Source Admin permissions are required to create and edit transformations.
> Source Read-only permissions are required to view transformations.

![create a transformation with the three-step wizard](../images/transformation_wizard.png)

### Step 1: Select the transformation type

> To create a Transformation, you first need to select which type of transformation you want to create. For each transformation type, Segment displays a description, use cases, and example payload. Current transformation types available in your Segment workspace include:

Reviewer: oxford comma, 👨‍🍳 💋

In this step, you can also choose to keep the Transformation disabled, so you can come back and edit it later. To update, enable, or disable a Transformation, click on the overflow menu and select **Edit Transformation**.

## Use Cases

Here's a list of Segment Transformations with some use case examples.

- **Rename an event:** Change an event name from `viewed_product` to `Product Viewed`.

- **Rename a property or trait:** Change the property name `revenue` to `total` for a specific destination.

- **Update a property value:** Use [Segment's Public API](https://docs.segmentapis.com/tag/Transformations){:target="_blank"} to transform the property `currency` to have the value `USD`. This transformation is in beta.

- **Add a new property name and assign a value:** If you want to create a new property and set a static value, use [Segment's Public API](https://docs.segmentapis.com/tag/Transformations){:target="_blank"} to create `new_property: static_value`. This transformation is in beta.

> <!-- hiding this transformation until it's available

Reviewer: Can you use liquid comments here? HTML comments still renders the content, but just prevents it from displaying on the page. Liquid comments block the commented section from being rendered to HTML.

```
{% comment %}
content
{% endcomment %}
```

- **Change property value casing:** Transform property value casing to lowercase, uppercase, or title case. For example, Transform the property value `united states` to `USA` to remain consistent with your data tracking.
--> 

> > note ""

Reviewer: ```suggestion
> info ""
```
